### ISSUE TITLE
Make it easier to find locations where ACA is available

### ISSUE THREAD

**Issue by charris-msft** [Azure/azure-cli#24286]
**Related command**
`az provider show locations`

**Is your feature request related to a problem? Please describe.**
ACA is not available in all locations, and it saves a lot of time if a user can see where it is available before attempting a deployment.
The current command is pretty unintuitive and hard to remember:
`az provider show -n Microsoft.App --query "resourceTypes[?resourceType=='managedEnvironments'].locations"`

**Describe the solution you'd like**
We should have something that is simple and easy to remember like this:
`az provider show locations -n ContainerApps`

There should also be a command/param to filter to regions that not only have ACA deployed, but also have available capacity:
`az provider show locations available -n ContainerApps`
or 
`az provider show locations -n ContainerApps -onlyAvailable`
(There is probably already a pattern for this, just providing a couple of examples.)

**Describe alternatives you've considered**
I've asked around and the ugly command above is the best anyone could think of.

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Comment by navba-MSFT** [comment]
Adding Service team to look into this.

**Comment by navba-MSFT** [comment]
@calvinsID Could you please look into this and provide an update ?

**Comment by zhoxing-ms** [comment]
@charris-msft I don't see the REST service supports finding location in [REST documents](https://learn.microsoft.com/en-us/rest/api/resources/providers
). Could you confirm that REST service supports this feature?
If so, please provide the REST docs and Swagger link

**Comment by charris-msft** [comment]
No, I can't confirm that. I'm not saying that it should work based on what we have currently implemented. Rather I believe we need to make whatever changes are required so that the command can be that simple across all providers.

**Comment by zhoxing-ms** [comment]
OK, I think this feature needs to be supported by the REST service first, so that multiple client tools can support it. 
And only if the REST service supports filtering locations can make those commands' performance be faster.

**Comment by charris-msft** [comment]
Makes sense. Can you forward it on to the correct team?

**Comment by zhoxing-ms** [comment]
@tjegbejimba Could you help take look at this issue or transfer it to the right person?
